### PR TITLE
Part1

### PR DESCRIPTION
**Pull request by tjaskula**
Finish D@D battle

### CODE REVIEW

File: DragonSlayer.cpp [hunk 1]
```diff
@@ -1,23 +1,39 @@
...
+            //DragonSlayers get a 10x boost when attacking dragons, from their attack item.
+
+            if (attackItem != nullptr)
+            {
+                //so they should USE their attack item before attacking the dragon... 
+                attackItem->use(this);
+                //note: items are single-use only, so you need to reset it after use.  
+                //look in the Character class for how the other item types are reset after use.
+                attackItem.reset();
+            }
```
Comment (matkatmusic) [inline DragonSlayer.cpp]: Use the attack item before the while() loop.
if you don't, you'll be performing this check every time this loop runs, which is redundant after the first usage.
- [x] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]
Fix all the warnings that occur during compilation

File: DragonSlayer.cpp (hunk 1 above)
Comment (tjaskula) [reply DragonSlayer.cpp]: I've made the change

**Review by matkatmusic — CHANGES_REQUESTED** [review]
The code still produces a ton of warnings. 
The fixes for those warnings are explained in the instructions. 
Please implement the fixes as explained in the instructions.
- [x] revise

**Review by matkatmusic — APPROVED** [review]

**Pull request merged**